test/e2e/metrics_test.go

```diff
@@ -0,0 +1,111 @@
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func testMetrics(t *testing.T) {
+
+	t.Run("Windows Exporter configuration validation", testWindowsExporter)
+	t.Run("Prometheus configuration validation", testPrometheus)
+}
+
+// testWindowsExporter deploys Linux pod and tests that it can communicate with Windows node's metrics port
+func testWindowsExporter(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+
+	// Need at least one Windows node to run these tests, throwing error if this condition is not met
+	require.Greater(t, len(gc.nodes), 0, "test requires at least one Windows node to run")
+
+	for _, winNode := range gc.nodes {
+		t.Run(winNode.Name, func(t *testing.T) {
+			// Get the node internal IP so we can curl it
+			winNodeInternalIP := ""
+			for _, address := range winNode.Status.Addresses {
+				if address.Type == core.NodeInternalIP {
+					winNodeInternalIP = address.Address
+				}
+			}
+			require.Greaterf(t, len(winNodeInternalIP), 0, "test requires Windows node %s to have internal IP", winNode.Name)
+
+			// This will curl the windows server. curl must be present in the container image.
+			linuxCurlerCommand := []string{"bash", "-c", "curl http://" + winNodeInternalIP + ":9182/metrics"}
```

Comment:
The `"9182/metrics"` can be replaced with `metricsPort` and `metricsPortName` variables